demos/preconditioning/demo_heat_rana.py.rst

```diff
@@ -0,0 +1,123 @@
+
+Rana/Howle/et al preconditioning
+====================================================
+
+This demo applies a method suggested in:
+
+Rana, Howle, Long, Meek, Milestone "A new block preconditioner for implicit Runge-Kutta methods for parabolic PDE problems," SISC 2021
+
+to our ongoing heat equation demonstration problem on :math:`\Omega = [0,10]
+\times [0,10]`, with boundary :math:`\Gamma`, giving rise to the weak form
+
+.. math::
+
+   (u_t, v) + (\nabla u, \nabla v) = (f, v)
+
+A multi-stage RK method applied to the heat equation gives a
+block-structured system.  The on-diagonal blocks are quite similar to
+what one obtains from a backward Euler discretization of the equation.
+
+With a 2-stage method, we have
+
+.. math::
+   
+   \left[ \begin{array}{cc} A_{11} & A_{12} \\ A_{21} & A_{22} \end{array} \right]
+   \left[ \begin{array}{c} k_1 \\ k_2 \end{array} \right]
+   &= \left[ \begin{array}{c} f_1 \\ f_2 \end{array} \right]
+
+The suggestion in the paper is to approximate the Butcher matrix A with
+with a triangular approximation.  For example, if A = LDU, then one could approximate A with LD or DU.  This gives rise to a block triangular preconditioner
+
+.. math::
+
+  P = \left[ \begin{array}{cc} \tilde{A}_{11} & 0 \\ \tilde_{A}_{21} & A_{22} \end{array} \right]
```

Comment:
Everything is, in principle, perturbed because you've approximated the Butcher matrix with a triangular one.  Possible A_{11} isn't perturbed if you're taking U of an LU factorization since the first row of U is unchanged, but there may be other situations where this doesn't happen.  So I'm tilde-ing `A_{22}`